### ISSUE TITLE
AnchorCheck plugin is broken

### ISSUE THREAD

**Issue by cjmayo** [linkchecker/linkchecker#460]
With current a977e4d7129450ba9fda8389724c80c1bde66883:

No warnings reported when this file is served as fragment.html and checked with `-r1` (or `-r2`):

```html
<html>
<body>
<a href="fragment.html">a</a>
<a href="fragment.html#one">a1</a>
<a href="fragment.html#two">a2</a>
<a href="fragment.html#three">a3</a>
<a href="fragment.html#four">a4</a>
<a href="fragment.html#five">a5</a>
</body>
</html>
```

Commenting out these two lines:

https://github.com/linkchecker/linkchecker/blob/a977e4d7129450ba9fda8389724c80c1bde66883/linkcheck/cache/urlqueue.py#L128-L129

then checking with `-r1` results in `5 warnings found` of the type:
```Warning    [None] Anchor `two' not found. Available anchors: -.```

checking with `-r2` gives `13 warnings found`, with several repeats.

9.3 and 9.4 do not report any warnings either. The commented out lines were introduced in 9.4:

eaa538c8 ("don't check one url multiple times", 2016-11-09)

**Comment by cjmayo** [comment]
Duplicate of #179.
